In [5]:
% some housekeeping stuff
%register_graphics_toolkit ("notebook");
register_graphics_toolkit ("plotly");
clear
format short e
% end of housekeeping

# More on Acids and CO2

In [ ]:
pKa1 = input("Enter pKa1 value: ");
pKa2 = input("Enter pKa2 value: ");
%pKa1=6.3; pKa2=10.3;
Ka1=10^-pKa1; Ka2=10^-pKa2;
pH=2:0.1:12; H=10.^-pH;

denominator=H.^2+Ka1*H+Ka1*Ka2;
alphaH2A=(H.^2)./denominator;
alphaHA=(Ka1*H)./denominator;
alphaA=1-alphaH2A-alphaHA;

plot(pH,alphaH2A,'linewidth',2,pH,alphaHA,'linewidth',2,pH,alphaA,'linewidth',2)
set(gca,'fontsize',12,'linewidth',2)
legend('H_2A','HA^-','A^{2-}')
title('Distribution Diagram for H2A')
xlabel('pH'); ylabel('\alpha','fontsize',16)

hold on
plot([pKa1 pKa1],[0 1],'k:','linewidth',2)
plot([pKa2 pKa2],[0 1],'k:','linewidth',2)
plot([2 12],[0.5 0.5],'k:','linewidth',2)

### metal ions as acids  
for $\mathrm{Al^{3+}}$ pka = 5.00.  
$\mathrm{Al^{3+} + H_2O \rightleftharpoons H^+ + AlOH^{2+}}$  
$\mathrm{Al(H_2O)_6^{3+} \rightleftharpoons Al(H_2O)_5OH^{2+} + H^+}$  

**pH of 0.01 M Al(NO<sub>3</sub>)<sub>3</sub>**  
reactions: $\mathrm{H_2O = H^+ + OH^-}$ Kw  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; $\mathrm{Al(NO_3)_3 \rightarrow Al^{3+} + 3NO_3^-}$  
$\mathrm{Al^{3+} + H_2O \rightleftharpoons AlOH^{2+} + H^+}$ Ka  
species: $\mathrm{H^+, OH^-, Al^{3+}, AlOH^{2+}, NO_3^-}$  
$[H^+] + 3[Al^{3+}] + 2[AlOH^{2+}] = [NO_3^-] + [OH^-]$  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;protonated&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;deprotonated

$\mathrm{\alpha_A = \frac{Ka}{Ka + [H^+]}}$  
$\mathrm{\alpha_{HA} = \frac{[H^+]}{Ka+[H^+]}}$  
$\mathrm{[H^+] + 3 (\frac{[H^+]Al_T}{Ka + [H^+]}) + 2(\frac{KaAl_T}{Ka + [H^+]})}$  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;$\mathrm{= NO_{3T}}$  
$\mathrm{Ka[H^+] + [H^+]^2 + 3[H^+]Al_T + 2KaAl_T}$

In [7]:
% insert the variables and values

AlT = 0.01; Ka = 10^(-5); NO3T = 0.03
%NO3T=3X0.01

% type out the quadratic constants (a, b and c)

a=1; b=Ka+3*AlT-NO3T; c=2*Ka*AlT-NO3T*Ka

% solve it. you do not need to change below this line

t=roots([a b c]); %finds the roots
t=t(imag(t)==0); %sets any imaginary roots to zero
% display the positive real root

t=t(t>0)

% take minus log in case that might be useful (pH for example)

pH=-log10(t)

% or this is the pOH

pOH=14-pH

NO3T = 0.030000


c = -1.0000e-07


t = 3.1127e-04


pH = 3.5069


pOH = 10.493


End of material for test #2
____

### CO<sub>2(g)</sub> in water  
$\mathrm{H_2CO_3 \rightleftharpoons H^+ + HCO_3^-}$ Ka<sub>1</sub>  
Carbonic Acid&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Bicarbonate  
$\mathrm{HCO_3^- \rightleftharpoons H^+ + CO_3^{2-}}$ Ka<sub>2</sub>  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Carbonate  
$\mathrm{Ka_1 = 10^{-6.3}}$  
$\mathrm{Ka_2 = 10^{-10.3}}$
![](moreacidsgraph.png)

### pH of rain water = ?  
$\mathrm{CO_{2(g)} \rightleftharpoons CO_{2(aq)}}$ KH  
Henry's Law  
$\mathrm{CO_{2(aq)} + H_2O \rightleftharpoons H_2CO_{3(aq)}}$  
add reactions:  
$\mathrm{CO_{2(g)} + H_2O \rightleftharpoons H_2CO_{2(aq)}}$  
$\mathrm{K_H = 10^{-1.5}}$(given on tests)  
species: $\mathrm{H^+, OH^-, HCO_3^-, CO_3^{2-}, H_2CO_3, CO_{2(g)}}$  
$\mathrm{P_{CO_2} = 10^{-3.5} atm}$  
charge balance  
$\mathrm{[H^+] = [OH^-] + [HCO_3^-] + 2[CO_3^{2-}]}$  
what is negligble? OH<sup>-</sup> acidic pH  
$\mathrm{[H^+] = [HCO_3^-]}$